lib/mj-single.js

```diff
@@ -549,6 +551,20 @@ function GetSpeech(result) {
 }
 
 //
+//  Creates the semantic tree for the current element and attaches it as JSON
+//  and XML.
+//
+function GetSemantic(result) {
+  if (!data.semantic) return;
+  if (!result.mml) {
+    data.mml = true ;
+    GetMML(result);
```

Comment:
Unfortunately, `GetMML` can't be called this way, since it is asynchronous, and could return before the result is available.  That is why `GetMML` and the other such functions are in the `HUB.Queue()` since the queue will wait until they complete before doing the next item in the queue.

So you probably should change [line 528](https://github.com/mathjax/MathJax-node/blob/stree_output/lib/mj-single.js#L528) in `GetMML` to not return if `data.semantic` is set.

Then I would move your `GetSemantic` call to before the `GetSpeech` call, because `GetSpeech` removes the `result.mml` value at [line 549](https://github.com/mathjax/MathJax-node/blob/stree_output/lib/mj-single.js#L549) if it is not to be returned.  (That should probably be done in a separate function or perhaps in `ReturnResult`.)